x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/crawler/create_crawler_index_api_logic.ts

```diff
@@ -15,23 +16,31 @@ export interface CreateCrawlerIndexArgs {
 }
 
 interface CreateCrawlerIndexRequest {
-  index_name: string;
   language: LanguageForOptimization;
 }
 
 export interface CreateCrawlerIndexResponse {
-  created: string; // the name of the newly created index
+  updated: string; // the name of the updated index
 }
 
 export const createCrawlerIndex = async ({ indexName, language }: CreateCrawlerIndexArgs) => {
-  const route = '/internal/enterprise_search/crawler';
-
-  const params: CreateCrawlerIndexRequest = {
+  const connRoute = '/internal/enterprise_search/connectors';
```

Comment:
You're putting too much interdependent logic in the browser where requests may fall or be left behind. What do you do if one request fails but the other succeeds? What happens if the browser refreshes or crashes before the second API call but after the first?

Instead, keep this as a single API call to `/internal/enterprise_search/crawler` and let that endpoint handle all of the logic. That way it'll be easier to modify later as well.